torch/export/dynamic_shapes.py

```diff
@@ -46,11 +46,13 @@ class _DimHint(Enum):
     - AUTO means automatic inference of shape (static or dynamic).
     - STATIC means static shape (always specialized).
     - DYNAMIC means dynamic, will error out if specialized.
+    - UNBACKED means allocating an unbacked symbol (with hint).
```

Comment:
Can you clarify the behavior of Dim.UNBACKED? In general I feel this will just add more confusion to users who want to specify dynamism.

My understanding is that this is like DYNAMIC but w/o 0/1 specialization, however below you did `mark_unbacked(..., strict=False)` meaning that it won't error if this is specialized. So that means the behavior is more like AUTO. But at the same time why can't AUTO just use mark_unbacked.